compiler/rustc_resolve/src/imports.rs

```diff
@@ -992,7 +992,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     && let Some(max_vis) = max_vis.get()
                     && !max_vis.is_at_least(import.expect_vis(), self.tcx)
                 {
-                    self.lint_buffer.buffer_lint(UNUSED_IMPORTS, id, import.span, fluent::resolve_glob_import_doesnt_reexport);
+                    self.no_reexports_imports.push(import);
```

Comment:
I delayed this process because the record of `a::﻿C` being used occurs at a 'late' stage.